internal/cmd/db.go

```diff
@@ -127,8 +130,64 @@ func getHost() string {
 	return host
 }
 
+type latMap struct {
+	id  string
+	lat *time.Duration
+}
+
 func locations(client *turso.Client) (map[string]string, error) {
 	settings, _ := settings.ReadSettings()
+	return readLocations(settings, client)
+}
+
+func latencies(client *turso.Client) (map[string]int, error) {
+	settings, _ := settings.ReadSettings()
+	locations, err := readLocations(settings, client)
+	if err != nil {
+		return nil, err
+	}
+
+	if latencies := settings.LatencyCache(); latencies != nil {
+		return latencies, nil
+	}
+
+	var wg sync.WaitGroup
+	latencies := make(map[string]int)
+	c := make(chan latMap)
+	for id := range locations {
+		for i := 0; i < 3; i++ {
+			wg.Add(1)
+			go func(id string) {
+				defer wg.Done()
+				dur := turso.ProbeLocation(id)
+				c <- latMap{id: id, lat: dur}
+
+			}(id)
+		}
+	}
+
+	go func() {
+		for kvp := range c {
```

Comment:
With buffered channel in line 156, this code could be in the main function after wg.Wait in line 183

Reply:
but what would be the main advantage?